### PR TITLE
refactoring and fix bug  using sharp 0.31.0 version's new features

### PR DESCRIPTION
**Pull request by tingxin**
1. remove svg text watermark
2. improve feature about auto resize  bigger watermark
3. fix bug: watermark was broken

### CODE REVIEW

File: source/new-image-handler/src/processor/image/watermark.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
+import { bool } from 'aws-sdk/clients/signer';
```
Comment (wchaws) [inline source/new-image-handler/src/processor/image/watermark.ts:1]: wrong import

File: source/new-image-handler/src/processor/image/watermark.ts [hunk 2]
```diff
@@ -228,58 +187,53 @@ export class WatermarkAction extends BaseImageAction {
...
-      }
-    }
     const pos = this.calculateImgPos(opt, metadata, markMetadata);
 
     const overlay = await this.extraImgOverlay(ctx, watermarkImg, markMetadata, opt, pos);
     ctx.image.composite([overlay]);
   }
 
-
   async mixedWaterMark(ctx: IImageContext, opt: WatermarkOpts): Promise<void> {
```
Comment (wchaws) [inline source/new-image-handler/src/processor/image/watermark.ts:215]: 文字阴影功能还有么？

File: source/new-image-handler/src/processor/image/watermark.ts (hunk 1 above)
Comment (tingxin) [reply source/new-image-handler/src/processor/image/watermark.ts]: fixed

File: source/new-image-handler/src/processor/image/watermark.ts (hunk 2 above)
Comment (tingxin) [reply source/new-image-handler/src/processor/image/watermark.ts:218]: added

File: source/new-image-handler/src/processor/image/watermark.ts [hunk 3]
```diff
@@ -23,17 +22,7 @@ export interface WatermarkOpts extends IActionOpts {
...
-interface WatermarkTextOpts extends IActionOpts {
-  width: number;
-  height: number;
-}
-
-interface WatermarkTextResizeOpts extends IActionOpts {
-  need: boolean;
-  width: number;
-  height: number;
+  halo: string; // shadow 颜色
```
Comment (wchaws) [inline source/new-image-handler/src/processor/image/watermark.ts:25]: why add this?

**Review by wchaws — APPROVED** [review]

**Pull request merged**